### PR TITLE
 bt-migrate: init at 0-unstable-2023-08-17 

### PR DESCRIPTION
**Pull request by ambroisie**
## Description of changes

Not a big fan of the new `unstable` naming...

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/bt/bt-migrate/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    mkdir -p $out/bin
+    cp BtMigrate $out/bin
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Torrent state migration tool";
+    homepage = "https://github.com/mikedld/bt-migrate?tab=readme-ov-file";
```
Comment (lolbinarycat) [inline pkgs/by-name/bt/bt-migrate/package.nix:57]:
```suggestion
    homepage = "https://github.com/mikedld/bt-migrate";
```

query parameter does nothing as far as i can tell

Comment (ambroisie) [reply pkgs/by-name/bt/bt-migrate/package.nix:57]: Oops, copy-paste error. Thank you.

File: pkgs/by-name/sq/sqlite_orm/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  propagatedBuildInputs = [
+    sqlite
+  ];
+
+  meta = with lib; {
+    description = "Light header only SQLite ORM";
+    homepage = "https://sqliteorm.com/";
+    license = licenses.agpl3Only; # MIT license is commercial
+    maintainers = with maintainers; [ ambroisie ];
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/sq/sqlite_orm/package.nix:31]: Set meta.platforms

File: pkgs/by-name/sq/sqlite_orm/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  propagatedBuildInputs = [
+    sqlite
+  ];
+
```
Comment (AndersonTorres) [inline pkgs/by-name/sq/sqlite_orm/package.nix:25]: Set strictDeps as true

File: pkgs/by-name/sq/sqlite_orm/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, cmake
+, fetchFromGitHub
+, sqlite
+, stdenv
+}:
+stdenv.mkDerivation (finalAttrs: {
```
Comment (AndersonTorres) [inline pkgs/by-name/sq/sqlite_orm/package.nix:7]: Nit:

```suggestion
}:

stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/di/digestpp/package.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    cp -r *.hpp algorithm/ detail/ $out/include/digestpp
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "C++11 header-only message digest library";
+    homepage = "https://github.com/kerukuro/digestpp";
+    license = licenses.unlicense;
+    maintainers = with maintainers; [ ambroisie ];
```
Comment (AndersonTorres) [inline pkgs/by-name/di/digestpp/package.nix:29]: meta.platforms

File: pkgs/by-name/sq/sqlite_orm/package.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, cmake
+, fetchFromGitHub
+, sqlite
+, stdenv
```
Comment (AndersonTorres) [inline pkgs/by-name/sq/sqlite_orm/package.nix:5]: Can stdenvNoCC be used?

File: pkgs/by-name/di/digestpp/package.nix [hunk 7]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, fetchFromGitHub
+, stdenv
```
Comment (AndersonTorres) [inline pkgs/by-name/di/digestpp/package.nix:3]: stdenvNoCC, maybe?

File: pkgs/by-name/bt/bt-migrate/package.nix [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Torrent state migration tool";
+    homepage = "https://github.com/mikedld/bt-migrate";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ ambroisie ];
+    mainProgram = "BtMigrate";
```
Comment (AndersonTorres) [inline pkgs/by-name/bt/bt-migrate/package.nix:59]: Set meta.platforms

File: pkgs/by-name/bt/bt-migrate/package.nix [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+  buildInputs = [
+    boost
+    cxxopts
+    fmt
+    jsoncons
+    pugixml
+    sqlite_orm
+  ];
+
```
Comment (AndersonTorres) [inline pkgs/by-name/bt/bt-migrate/package.nix:36]: set strictDeps as true

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sq/sqlite_orm/package.nix (hunk 2 above)
Comment (ambroisie) [reply pkgs/by-name/sq/sqlite_orm/package.nix:31]: Completely zapped them, sorry.

File: pkgs/by-name/sq/sqlite_orm/package.nix (hunk 3 above)
Comment (ambroisie) [reply pkgs/by-name/sq/sqlite_orm/package.nix:25]: Curious, is that becoming the new normal on nixpkgs or just your personal preference? To know whether I should encourage its use in reviews.

File: pkgs/by-name/sq/sqlite_orm/package.nix (hunk 6 above)
Comment (ambroisie) [reply pkgs/by-name/sq/sqlite_orm/package.nix:5]: We need `CC` for CMake.

**Pull request merged**